lib/api/workloadendpoint.go

```diff
@@ -36,13 +36,17 @@ type WorkloadEndpointMetadata struct {
 	Name string `json:"name,omitempty" validate:"omitempty,namespacedname"`
 
 	// The name of the workload.
-	Workload string `json:"workload,omitempty" valid:"omitempty,namespacedname"`
+	Workload string `json:"workload,omitempty" validate:"omitempty,namespacedname"`
+
+	// ActiveInstanceID contains an identifier for the container networked for
```

Comment:
Since this is optional and not actually a unique identifier, can you move this to be above the Labels field?

I don't like this wording, and it doesn't actually say what it does with it.  It should cover the following:

-  It isn't a unique identifier for the endpoint in that you cannot have multiple endpoints with the same Name, Workload, Orchestrator and Node and different ActiveInstanceIDs.
-  When you update/apply/create the endpoint, the endpoint uniquely identified by Node, Orchestrator, Workload and Name is updated to have the specified ActiveInstanceID.
-  When an ActiveInstanceID is store for the endpoint it is used to prevent deletion of the endpoint unless the delete request contains the same ActiveInstanceID  (+ rules about what happens when it isn't specified).